### PR TITLE
Modified the Eigen.hpp.in file to set the proper includes

### PR DESCRIPTION
**Pull request by whaeck**
I also added the sparsematrixplugin.hpp file for the operator() on sparse matrices.

I did encounter an issue. I had to copy the Eigen.hp.in file into another Eigen.hp file so that the #include "Eigen/Eigen.hpp" because cmake doesn't take off the .in extension when it copies to the include directory. This should be corrected but I do not know how.

### CODE REVIEW

**Comment by apmccartney** [comment]
It's proper that the `Eigen/Eigen.hpp` header was not found. Project headers are copied into the highest level of the header directory in njoy's build system, viz. you can include the file with the command:
```cpp
#include "Eigen.hpp" 
/* as opposed to 
#include "Eigen/Eigen.hpp"
*/
```
For a inspectable example, check out and build the math project. Within the build tree, you'll notice an `include` directory that's constructed.

The fact that `Eigen.hpp.in` is copied with the rest of the project is an (innocuous) bug in the quick-and-dirty Eigen integration. I'll fix that early next week.

File: Eigen.hpp [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
```
Comment (apmccartney) [inline Eigen.hpp]: This file ought not be included in the pull request. It's generated from `Eigen.hpp.in` at CMake invocation.

Comment (whaeck) [inline Eigen.hpp]: I removed the file. There are still some subtle features of the build system I do not grasp entirely. I'll get there in time :-)

Comment (apmccartney) [inline Eigen.hpp]: Once that update is pushed, I'll merge the pull request

**Comment by whaeck** [comment]
The file has been removed. I have changed the include statements in the Matrix, Vector and SparseMatrix include files in my local branch and everything compiles nicely now. Thanks Austin.

This can now be merged.
